ESP8266Interface.cpp

```diff
@@ -64,20 +67,31 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
-    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
-    
-    if (!_esp.reset()) {
-        return NSAPI_ERROR_DEVICE_ERROR;
-    }   
- 
+    size_t ssid_length = strlen(ap_ssid);
+    size_t pass_length = strlen(ap_pass);
+
+    if (ssid_length == 0 || ssid_length > ESP8266_SSID_MAX_LENGTH) {
```

Comment:
Why are these checks required? They're reading from the member arrays, which are already shorter than the maximum, and it seems both min and max sizes were checked when setting them in set_credentials.

On general principle, returning a "PARAMETER" error in a function with no parameters is a bit nonsensical.

Reply:
On general principle, returning a "PARAMETER" error in a function with no parameters is a bit nonsensical.
And still a developer might call this function without calling set_credentials() first. OK, check against max length is nonsense. But maybe would still need to check we have a SSID.

Reply:
Ah, so there's no default SSID, so default connect() alone can fail. 

Are we happy with that, generally? Should there be a standard JSON option for default SSID or something?

I note there is a "NSAPI_ERROR_NO_SSID". Seems appropriate.